src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/AssetCache.cs

```diff
@@ -178,83 +220,158 @@ public static IEnumerable<T> EnumerateAllComponentsInScenesAndPrefabs<T>()
+    internal abstract class AssetCache<TAsset> : AssetCache, IAssetCache
         where TAsset : UnityEngine.Object
     {
-        [SerializeField]
-        private TAssetEntry[] assets = null;
+        const string assetContentExtension = ".asset";
 
-        private Dictionary<AssetId, TAssetEntry> lookupByAssetId;
-        private Dictionary<TAsset, TAssetEntry> lookupByAsset;
+#pragma warning disable 414
+        [SerializeField]
+        private NameEntry[] assets = null;
+#pragma warning restore 414
 
-        protected IDictionary<AssetId, TAssetEntry> LookupByAssetId
+        private Dictionary<string, List<AssetId>> LookupByName
         {
             get
             {
-                if (lookupByAssetId == null)
+                if (lookupByName == null)
                 {
-                    if (assets == null)
-                    {
-                        lookupByAssetId = new Dictionary<AssetId, TAssetEntry>();
-                    }
-                    else
+                    lookupByName = new Dictionary<string, List<AssetId>>();
+                    if (assets != null)
                     {
-                        lookupByAssetId = assets.Where(a => a.Asset != null).ToDictionary(a => a.AssetId);
+                        for (int i = 0; i < assets.Length; i++)
+                        {
+                            lookupByName[assets[i].Name] = assets[i].Ids.ToList();
+                        }
                     }
                 }
-                return lookupByAssetId;
-            }
-        }
 
-        protected IDictionary<TAsset, TAssetEntry> LookupByAsset
-        {
-            get
-            {
-                if (lookupByAsset == null)
-                {
-                    if (assets == null)
-                    {
-                        lookupByAsset = new Dictionary<TAsset, TAssetEntry>();
-                    }
-                    else
-                    {
-                        lookupByAsset = assets.Where(a => a.Asset != null).ToDictionary(a => a.Asset);
-                    }
-                }
-                return lookupByAsset;
+                return lookupByName;
             }
         }
 
+#pragma warning disable 414
+        private Dictionary<string, List<AssetId>> lookupByName;
+        private Dictionary<AssetId, AssetCacheEntry> lookupByAssetId = new Dictionary<AssetId, AssetCacheEntry>();
+        private Dictionary<TAsset, AssetCacheEntry> lookupByAsset = new Dictionary<TAsset, AssetCacheEntry>();
+#pragma warning restore 414
+
         public TAsset GetAsset(AssetId assetId)
         {
             using (StateSynchronizationPerformanceMonitor.Instance.IncrementEventDuration(this.GetType().Name, "GetAsset"))
             {
-                TAssetEntry assetEntry;
-                if (LookupByAssetId.TryGetValue(assetId, out assetEntry))
+                if (assetId == null ||
+                assetId == AssetId.Empty ||

```

Comment:
>assetId == AssetId.Empty || [](start = 16, length = 27)

Nit: I think there are some missing indents here (i.e. bring assetId on the next level to at least where assetId is on the preceding one (and same for the next line).